lib/src/impl/paginated_result.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:ably_flutter_plugin/ably.dart';
+import 'package:ably_flutter_plugin/src/impl/message.dart';
+import 'package:ably_flutter_plugin/src/impl/platform_object.dart';
+import 'package:ably_flutter_plugin/src/spec/spec.dart' as spec;
+
+class PaginatedResult<T> extends PlatformObject
+    implements spec.PaginatedResultInterface<T> {
+
+  /// stores page handle created by platform APIs
+  ///
+  /// handle is updated after creating an instance as they are received
+  /// as 2 different properties of AblyMessage, and this instance is
+  /// instantiated by the codec. So the code invoking platform method
+  /// is bound to update this [_pageHandle]
+  ///
+  /// [PaginatedResult.fromAblyMessage] will act as a utility to update
+  /// this property. See [next] and [first] for usages
+  int _pageHandle;
+
+  /// items contain page of results
+  @override
+  List<T> items;
```

Comment:
This should probably be a private field with a public getter

Reply:
@zoechi addressed at 5df822ddbe75e9bd504bb69471273002d38bedde